src/Database/Type/FloatType.php

```diff
@@ -65,18 +65,18 @@ public function __construct($name = null)
      *
      * @param string|resource $value The value to convert.
      * @param \Cake\Database\Driver $driver The driver instance to convert with.
-     * @return string|resource
+     * @return string|null
      */
     public function toDatabase($value, Driver $driver)
     {
         if ($value === null || $value === '') {
             return null;
         }
         if (is_array($value)) {
-            return 1;
+            return '1';
         }
 
-        return (float)$value;
+        return sprintf('%F', (float)$value);
```

Comment:
First of all, I don't oppose this change. It could be better than now. Strictly, in MySQL, such big decimal values cannot be passed to the database as strings, [until MySQL 5.7.8](https://bugs.mysql.com/bug.php?id=72056). But now it works.

But, in my opinion, `decimal` shouldn't be truncated in any case. (In SQL server `numeric` is functionally equivalent to `decimal`. And, in my above comment, `number` is a typo. Sorry)

When people use `decimal` instead of `float`, they would need to store important data (e.g. money) to the field. They want to avoid to lose precision of the value. However, if we map `deciaml` to `float`, it would disappoint their expectation.

Now I think `decimal` should be mapped to `string` instead of `float`. It should be done in 3.4, as it would break BC. `DecimalType` class might be unnecessary. At first I thought we should create the class. But on second thoughts, there is not much we can do in the class in PHP.